### PR TITLE
Move Ordebook to autopilot::infra

### PR DESCRIPTION
**Pull request by sunce86**
# Description
A follow-up for https://github.com/cowprotocol/services/pull/2246

Moves `Orderbook` to `infra` where it belongs <3

## How to test
Existing tests.

### CODE REVIEW

File: crates/autopilot/src/infra/mod.rs [hunk 1]
```diff
@@ -1,2 +1,5 @@
 pub mod blockchain;
+pub mod orderbook;
 pub mod persistence;
+
+pub use orderbook::Orderbook;
```
Comment (fleupold) [inline crates/autopilot/src/infra/mod.rs:5]: I'd make this `infra::shadow::Orderbook`

File: crates/autopilot/src/infra/orderbook/auction.rs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+use crate::{
+    domain,
+    infra::{self, persistence::auction::dto},
+};
+
+impl infra::Orderbook {
```
Comment (fleupold) [inline crates/autopilot/src/infra/orderbook/auction.rs:6]: Let's merge with parent impl for a single view on all it's parts.

**Review by fleupold — APPROVED** [review]
Looks good, but let's please not use impl extensions unless necessay and make it explicit that this piece of infra is only used for the shadow autopilot.

**Review by squadgazzz — APPROVED** [review]

File: crates/autopilot/src/infra/mod.rs (hunk 1 above)
Comment (sunce86) [reply crates/autopilot/src/infra/mod.rs]: fixed, referred with `infra::shadow::Orderbook` in the Runloop

File: crates/autopilot/src/infra/orderbook/auction.rs (hunk 2 above)
Comment (sunce86) [reply crates/autopilot/src/infra/orderbook/auction.rs]: merged

**Pull request merged**